### ISSUE TITLE
Topics: Close without saving button (X) not working 

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1044]
Will make a new issue for this one so the complete bits in [this related issue](https://github.com/citizenos/citizenos-fe/issues/931#event-12150692567) can remain closed :) 

@ilmartyrk I just tried out the X for 'Close without saving', and it didn't work for me. A couple of issues I found: 

1. **Topic not closing when clicking X, and one edit remaining** 

I added the word edit to the topic text in a draft topic, and I changed the title from 'Test create' to 'Test edit':

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/9c7a125b-3832-42af-8e19-4824553d875c)

After clicking the X, the topic stayed open (it should close) and the title stayed the same. The additional word in the topic text did disappear though.  

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/0d740526-923d-4812-94c7-20ec8938b1d2)

This means the user gets stuck in the topic, and the only thing they can do is publish or save as draft, as every time they try to exit in another way (i.e. go to dashboard, etc.) the interrupt process pop-up appears but the topic doesn't close. 

EDIT: for me the X did just work for a topic that was already published - I was able to click X, accept the pop-up and it closed. So this particular problem seems to only be in draft mode? 

2. **One edit remaining, and taken back to dashboard not to the topic.** 

I tried it on desktop with an already published test topic. I added 'test edit' to the title and the topic text: 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/a6981c01-9c00-4361-83f1-5e75341c9423)

When I clicked the X, it took me to the dashboard (it should take me back to the topic as that's where I came from), and the 'test edit' had gone from the title but not from the text: 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/7fbf5665-bc88-445c-80b6-d1387ddbd3ab)

My suggestion as this seems a complicated one with lots of room for bugginess, maybe once Ilmar has fixed this then @ssin1901 might test more thoroughly than I'm able to (i.e. on new topics in create flow, on draft topics in edit flow, and on published topics in edit flow, and having edited various different things not just text - including settings), to check if there are any other issues coming up? 

And just to be clear for Sara if you do test it more thoroughly, am copying the intended UX here:

_The 'X' buttons in all cases should close the flow and not save whatever they've done since they opened the create or edit flow, but only after they have confirmed via the 'Interrupt process' pop-up which includes the sentence about changes not being saved. In the create flow, after this they should be taken to the dashboard. In the edit flow, after this they should be taken back to the topic._

(And the interrupt process pop-up should always appear after the click X, to be sure they want to exit without saving)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish @ssin1901 I believe it got fixed during the launch process and fixes after that

**Comment by ssin1901** [comment]
Seems to be working (screenshot1). Closing now.
<img width="1440" alt="Screenshot 2024-03-21 at 11 04 23" src="https://github.com/citizenos/citizenos-fe/assets/72747107/a9ed8fba-1064-451d-a561-a3373af30eab">

**Comment by BeccaMelhuish** [comment]
Reopening as not working yet I'm afraid :) 

I tried point 1 above again, doing exactly as I had done there (adding text to a draft and then interrupting process via the X) and now the edits did disappear as they should, but the topic remained open in edit mode. I.e. clicking the X didn't close it, it took me back to the same place I was in (see screenshot below), the only result of clicking X was to delete the edits I'd made. So as well as removing the edits, it needs to close and take the user back to the dashboard. 

Same both on mobile (Android, Chrome) and on desktop (Windows, Chrome).

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/df284ae6-6039-4a35-947a-1be995eb2e06)

For point 2, again it's closer, but not quite working. I did the same as I did above (edited a published topic, but clicked the X instead of saving changes). It now takes me back to the topic, which is good, but one of the words I had added when in edit mode was still there (but should have been removed). I had also added a word to the title, and that was removed, but not the one in the topic text: 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/d4a76180-5903-48e0-a382-5ca9fcc2182a)

This second one I only tried on desktop.

**Comment by ilmartyrk** [comment]
@BeccaMelhuish I need more information as I cannot replicate it.
What browser and OS were you using, did it happen on live or testing app and also what is the exact flow.
Create discussion topic ? Edit draft? Create voting topic? :)

**Comment by BeccaMelhuish** [comment]
@ilmartyrk here is a screen recording for the first one, on my laptop (Windows, Chrome) but the same thing happened when I tried it on mobile (Android, Chrome). This is in live. 

https://www.loom.com/share/7f240493284e4bd599fda579ecefaf85?sid=9e65c99a-aff8-49a9-8135-49e81a590e14

And here is one for the second issue, which actually works perfectly in this video, so isn't showing the problem only the flow: 

https://www.loom.com/share/be56aff4e7b74afbb18cf3da6dbceba1?sid=3da227f7-8f43-4ded-bbd8-bc1d20642a2d

But when I did the same thing earlier (15 min ago), the word editing remained when it shouldn't have: 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/ea0a56eb-34bf-4696-8a96-bec9bee59567)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish oh now I see what is going on with issue 1. it is easy fix, but short explanation on what happened. Previously when you edited a draft topic and closed it, it redirected to topic read view, but it should not have been allowed for draft topics, I fixed that so if a user opens draft topic in read view from URL they get redirected to edit mode.

When you start creating a new topic and decide to move away the topic will get deleted, but for edit mode it will redirect to topic read view as we don't delete the topic and together with previous description is what happened with draft topics. Will post a fix soon

**Comment by ilmartyrk** [comment]
@BeccaMelhuish @ssin1901 posted a fix, 1. should be fixed, 2. added a little delay before topic content is reloaded so it would be up to date.  The edits are actually not saved, but as it is a technical interaction between API and Etherpad to make it work, there is some delay when the added changes might be visible.

**Comment by ssin1901** [comment]
@BeccaMelhuish if I have understood correctly – it should be tested in Draft mode?

**Comment by BeccaMelhuish** [comment]
@ssin1901 first problem was in draft mode but second in published mode, the screen recordings above should make it clearer though hopefully :) 

Might also be good to test if changing anything else such as image, attachments, settings, etc. get discarded after clicking the X (as they should)? I only tested quickly with adding some words..

**Comment by ssin1901** [comment]
@BeccaMelhuish thanks a lot for the screen recordings! Looks an important issue, but let me confirm because I need to write a few new tests to test all the functionalities.

_The 'X' buttons in all cases should close the flow and not save whatever they've done since they opened the create or edit flow, but only after they have confirmed via the 'Interrupt process' pop-up which includes the sentence about changes not being saved. In the create flow, after this they should be taken to the dashboard. In the edit flow, after this they should be taken back to the topic.

(And the interrupt process pop-up should always appear after the click X, to be sure they want to exit without saving)_

So to be clear – if a user edits the topic but clicks x, whatever edits they make should disappear only after clicking x BUT if they don't click x and simply navigate away from the topic, then everything should be autosaved?

Further - interrupt process in Create mode should lead to Dashboard, in edit mode should lead to the topic that was interrupted?

**Comment by BeccaMelhuish** [comment]
@ssin1901 all sounds correct to me, thank you! :)
